helix-term/src/ui/editor.rs

```diff
@@ -74,6 +78,113 @@ impl EditorView {
+            Left,
+            Center,
+        }
+
+        /// Declare the welcome screen declaratively using this macro
+        /// It makes it easy to get the longest line in the center and the left, without
+        /// having to iterate over the `Vec<_>` again.
+        macro_rules! welcome {
+            (
+                $([$align:ident] $line:expr, $(if $cond:expr;)?)* $(,)?
+            ) => {{
+                let mut lines = vec![];
+                let mut longest_left = 0;
+                let mut longest_center = 0;
+                $(
+                    let line = Spans::from($line);
+                    let width = line.width();
+                    lines.push((line, Align::$align));
+                    match Align::$align {
+                        Align::Left => longest_left = longest_left.max(width),
+                        Align::Center => longest_center = longest_center.max(width),
+                    }
+                )*
+                (lines, longest_left, longest_center)
+            }};
+        }
+
+        let (lines, len_of_longest_left, len_of_longest_center) = welcome! {
+            [Center] vec!["helix ".into(), Span::styled(VERSION_AND_GIT_HASH, theme.get("comment"))],
+            [Left] "",
+            [Center] Span::styled(
+                "A post-modern modal text editor",
+                theme.get("ui.text").add_modifier(Modifier::ITALIC),
+            ),
+            [Left] "",
+            [Left] vec![
+                Span::styled(":tutor", theme.get("markup.raw")),
+                Span::styled("<enter>", theme.get("comment")),
+                "       learn helix".into(),
+            ],
+            [Left] vec![
+                Span::styled(":theme", theme.get("markup.raw")),
+                Span::styled("<tab>", theme.get("comment")),
+                "         choose a theme".into(),
+            ],
+            [Left] vec![
+                Span::styled("<space>e", theme.get("markup.raw")),
+                "            file explorer".into(),
+            ],
+            [Left] vec![
+                Span::styled("<space>?", theme.get("markup.raw")),
+                "            see all commands".into(),
+            ],
+            [Left] vec![
+                Span::styled(":config-open", theme.get("markup.raw")),
+                Span::styled("<enter>", theme.get("comment")),
+                " configure helix".into(),
+            ],
+            [Left] vec![
+                Span::styled(":quit", theme.get("markup.raw")),
+                Span::styled("<enter>", theme.get("comment")),
+                "        quit helix".into(),
+            ],
+            [Left] "",
+            [Center] vec![
+                Span::styled("docs: ", theme.get("ui.text")),
+                Span::styled("docs.helix-editor.com", theme.get("markup.link.url")),
+            ],
+        };
+
+        // how many total lines there are in the welcome screen
+        let lines_count = lines.len();
```

Comment:
Maybe define these counts as u16 instead of littering the code below with casts?